A3A/addons/core/functions/AI/fn_fastropeVTOL.sqf

```diff
@@ -0,0 +1,209 @@
+#include "..\..\script_component.hpp"
+FIX_LINE_NUMBERS()
+
+params ["_veh", "_groupX", "_positionX", "_posOrigin", "_heli"];
+
+private _vehType = typeOf _veh;
+
+_veh setVehicleRadar 1;
+
+private _reinf = if (count _this > 5) then {_this select 5} else {false};
+
+private _xRef = 2;
+private _yRef = 1;
+private _landpos = [];
+private _dist = if (_reinf) then {30} else {100 + random 100};
+
+while {true} do
+	{
+ 	_landpos = _positionX getPos [_dist,random 360];
+ 	if (!surfaceIsWater _landpos) exitWith {};
+	};
```

Comment:
```suggestion
private _landPos = [nil, ["water"]] call BIS_fnc_randomPos;
```

Stop using while loops, try this and see if it works

Reply:
This is how it was in the original